### ISSUE TITLE
Wrong C# Stacktrace when exception occurs in _Ready when called from AddChild

### ISSUE THREAD

**Issue by WizzardMaker** [godotengine/godot#30208]
**Godot version:** 3.1.1 Mono

**OS/device including version:** Windows 10

**Issue description:**
Whenever an exception occurs inside a `_Ready()` function the stacktrace isn't taken further than the `AddChild()` function

Say, I have 2 scenes. One loading scene and one game scene. When I finish loading and want to change the scene to the game scene I load the scene with `ResourceLoader.Load()`, `QueueFree()` the root node and `AddChild()` the loaded game scene.
The problem happens when any script run inside the second scene throws an exception in `_Ready()`. This exception has a "broken" stacktrace.

**Steps to reproduce:**
1. Create two scenes
2. Create a script that loads a scene and adds it with `AddChild()` and add that script to the root of one scene
3. Create a script that throws an exception in `_Ready()` and add that to the root of the other scene
4. Run the project

**Minimal reproduction project:**
Example:
[ExceptionTest.zip](https://github.com/godotengine/godot/files/3343179/ExceptionTest.zip)

Example Stacktrace:
```
Unhandled Exception:
System.Exception: Exception of type 'System.Exception' was thrown.
  at (wrapper managed-to-native) Godot.NativeCalls.godot_icall_2_378(intptr,intptr,intptr,bool)
  at Godot.Node.AddChild (Godot.Node node, System.Boolean legibleUniqueName) [0x00000] in /tmp/build_GodotSharp/GodotSharp/ObjectType/Node.cs:389
  at Loader._Process (System.Single delta) [0x0003f] in C:\Jonas\ExceptionTest\Loader.cs:17
```
Expected Stacktrace:
```
System.Exception: Exception of type 'System.Exception' was thrown.
  at ExceptionThrower.Ready () [-] in C:\Jonas\ExceptionTest\ExceptionThrower.cs:13
  at Godot.Node.AddChild (Godot.Node node, System.Boolean legibleUniqueName) [0x00000] in /tmp/build_GodotSharp/GodotSharp/ObjectType/Node.cs:389
  at Loader._Process (System.Single delta) [0x0003f] in C:\Jonas\ExceptionTest\Loader.cs:17
```

**Comment by giulianob** [comment]
I have the same issue and it makes debugging very difficult sometimes. Exception happened in `_Ready()` but this is all I get:
![image](https://user-images.githubusercontent.com/92303/62164385-520a2a00-b2ea-11e9-9854-ddb7a3ffe9e5.png)

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

**Comment by akien-mga** [comment]
Seems fixed as of 3.2.4 beta 2:
```
E 0:00:02.347   void ExceptionThrower._Ready(): System.Exception: Exception of type 'System.Exception' was thrown.
  <C++ Error>   Unhandled exception
  <C++ Source>  /home/akien/tmp/godot/bug/ExceptionTest/ExceptionThrower.cs:13 @ void ExceptionThrower._Ready()()
  <Stack Trace> ExceptionThrower.cs:13 @ void ExceptionThrower._Ready()()
                :0 @ ()
                Node.cs:526 @ void Godot.Node.AddChild(Godot.Node , Boolean )()
                Loader.cs:17 @ void Loader._Process(Single )()
```
